go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -2445,6 +2445,9 @@ func newGateway(
 			gwBridge.eipMarkIPs = gw.bridgeEIPAddrManager.GetCache()
 		}
 		gw.nodeIPManager = newAddressManager(nodeName, kube, mgmtPort, watchFactory, gwBridge)
+		if gw.nodeIPManager == nil {
```

Comment:
do you think it makes sense to refactor `newAddressManager to return an error? shouldn't be too difficult